packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/__tests__/useUploadFiles.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import { Storage, UploadTask } from '@aws-amplify/storage';
+
+import { FileStatus, StorageFile } from '../../../types';
+import { useUploadFiles, UseUploadFilesProps } from '../useUploadFiles';
+import { waitFor } from '@testing-library/react';
+
+jest.mock('@aws-amplify/storage');
+
+const mockUploadingFile: StorageFile = {
+  id: 'uploading',
+  status: FileStatus.UPLOADING,
+  progress: 0,
+  error: '',
+  isImage: false,
+  key: '',
+};
+
+const imageFile = new File(['hello'], 'hello.png', { type: 'image/png' });
+
+const mockQueuedFile: StorageFile = {
+  id: 'queued',
+  status: FileStatus.QUEUED,
+  progress: 0,
+  error: '',
+  isImage: false,
+  key: 'key',
+  file: imageFile,
+};
+
+const mockSetUploadingFile = jest.fn();
+const mockSetUploadProgress = jest.fn();
+const mockSetUploadSuccess = jest.fn();
+const mockOnUploadError = jest.fn();
+const props: Omit<UseUploadFilesProps, 'files'> = {
+  accessLevel: 'public',
+  maxFileCount: 2,
+  setUploadingFile: mockSetUploadingFile,
+  setUploadProgress: mockSetUploadProgress,
+  setUploadSuccess: mockSetUploadSuccess,
+  onUploadError: mockOnUploadError,
+};
+
+const storageOutput: UploadTask = {
+  resume: jest.fn(),
+  pause: jest.fn(),
+  percent: 100,
+  isInProgress: false,
+};
+
+describe('useUploadFiles', () => {
+  afterEach(() => jest.clearAllMocks());
+
+  it('should upload all queued files', async () => {
+    (Storage.put as jest.Mock).mockResolvedValue(storageOutput);
```

Comment:
Can we create a spy for `Storage.put` rather than directly mocking?

Reply:
What is the concern here?

Reply:
Idk, imo we really shouldn't be using mocks unless we absolutely need to (in particular with 3rd party code) but don't feel too strongly. Up to you :)

Reply:
Valid, but since we already have coverage for `Storage.put` using spies in the uploadfile tests, think it's okay to mock in this particular case. These tests as I view them is about mocking certain behavior from Storage.put (success or failure) and making sure this hook behaves accordingly.